instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/library-autoconfigure/src/main/java/io/opentelemetry/instrumentation/log4j/contextdata/v2_17/OpenTelemetryContextDataProvider.java

```diff
@@ -25,6 +26,19 @@
  * #supplyContextData()} is called when a log entry is created.
  */
 public class OpenTelemetryContextDataProvider implements ContextDataProvider {
+
+  private boolean configuredResourceAttributeAccessible;
```

Comment:
change this to a static final field and change what you have in constructor into a static method that returns a boolean and use it to initialize the field